src/corecel/math/Algorithms.hh

```diff
@@ -648,8 +648,9 @@ CELER_WRAP_MATH_FLOAT_DBL_1(, sinpi)
 CELER_WRAP_MATH_FLOAT_DBL_1(, cospi)
 CELER_WRAP_MATH_FLOAT_DBL_PTR_2(, sincospi)
 CELER_WRAP_MATH_FLOAT_DBL_PTR_2(, sincos)
-#elif __APPLE__ && !defined(__CLING__)
-// Apple defines __sinpi, __sinpif, __sincospi, ...
+#elif (__MAC_OS_X_VERSION_MIN_REQUIRED > 1090) \
+    || (__IPHONE_OS_VERSION_MIN_REQUIRED > 70000)
```

Comment:
iphone, really?  humm ... I guess soon enough its gpu will be strong enough to run Celeritas :) :)

Reply:
I copied the condition from Apple's `math.h`... might as well be "exactly right", no? 😉

Reply:
Yes but (especially because it is apple) it might change without warning :(

Reply:
eg.  The version on my system is:
```
#if (defined __MAC_OS_X_VERSION_MIN_REQUIRED && __MAC_OS_X_VERSION_MIN_REQUIRED < 1090) || \
    (defined __IPHONE_OS_VERSION_MIN_REQUIRED && __IPHONE_OS_VERSION_MIN_REQUIRED < 70000)
```

Reply:
Well right, but the "else" conditional is the one that has sincospi. So if the `min required` is undefined then we have the condition  0 > 1090, etc., which is a nice fallback for the macros going undefined.